react/src/pages/datasets/DatasetInfoDialog.tsx

```diff
@@ -0,0 +1,235 @@
+import { Dialog, DialogTitle, DialogContent, Paper, Typography,Grid, IconButton, Collapse, Button, List, ListItem, ListItemText, ListItemIcon, Box } from '@material-ui/core';
+import { Close, ShowChart, ExpandLess, ExpandMore } from '@material-ui/icons';
+import { Dataset, Analysis, Sample } from '../utils';
+
+const useStyles = makeStyles(theme => ({
+    datasetInfo: {
+        paddingLeft: theme.spacing(2),
+        paddingRight: theme.spacing(2),
+    },
+    paper: {
+        padding: theme.spacing(2),
+        marginBottom: theme.spacing(3),
+    },
+    listPaper: {
+        padding: theme.spacing(1),
+        margin: theme.spacing(1)
+    },
+    root: {
+        margin: 0,
+        padding: theme.spacing(2),
+    },
+    closeButton: {
+        position: 'absolute',
+        right: theme.spacing(1),
+        top: theme.spacing(1),
+        color: theme.palette.grey[500],
+    },
+    grid: {
+        paddingBottom: theme.spacing(2)
+    },
+    box: {
+        padding: theme.spacing(2),
+        margin: theme.spacing(1)
+    }
+}));
+
+interface DialogTitleProps {
+    id: string;
+    children: React.ReactNode;
+    onClose: () => void;
+}
+
+const DialogHeader = ((props: DialogTitleProps) => {
+    const { children, onClose, ...other } = props;
+    const classes = useStyles();
+    return (
+        <DialogTitle disableTypography className={classes.root} {...other}>
+            <Typography variant="h6">{children}</Typography>
+            {onClose ? (
+                <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
+                    <Close />
+                </IconButton>
+            ) : null}
+        </DialogTitle>
+    );
+});
+
+const getInfo = (subtitle: string, value: string | number | null) => {
+    let val: string;
+    if (typeof value === 'number')
+        val = value.toString();
+    else if (typeof value === 'string')
+        val = value;
+    else
+        val = "";
+
+    return (
+        <Typography variant="body1" gutterBottom><b>{subtitle}:</b> {val}</Typography>
+    );
+}
+
+interface DialogProp {
+    open: boolean,
+    dataset_id: string,
+    onClose: (() => void),
+}
+
+export default function DatasetInfoDialog({ dataset_id, open, onClose }: DialogProp) {
+    const classes = useStyles();
+    const labeledBy = "dataset-info-dialog-slide-title";
+
+    const [dataset, setDataset] = useState<Dataset>();
+    const [analyses, setAnalyses] = useState<Analysis[]>([]);
+    const [sample, setSample] = useState<Sample>();
+    const [moreDetails, setMoreDetails] = useState(false);
+    const [showAnalysis, setShowAnalysis] = useState<boolean[]>([]);
+
+    function clickAnalysis(index: number) {
+        try {
+            // toggle
+            setShowAnalysis(
+                showAnalysis.map((val, i) => index === i ? !val : val)
+            );
+        } catch (error) {
+            console.error(error);
+        }
+    }
```

Comment:
What is the purpose of the try-catch here?

Reply:
I think it was to catch an IndexError, but I changed the code inside the try block since then, so it's no longer necessary.